### DISCUSSION TITLE
Why would these commands fail within [dos]

### DISCUSSION THREAD

**Discussion by ChiKwasi** [ImageMagick/ImageMagick#7453] [Digital Image Processing] [answered]
I'm trying to create thumbnails from uploaded images.
Some have extreme aspect ratios, but that's OK

I resize to a width, and then crop from the center outward to fill a 215x280 area

Here's my issue.

This works fine from the command line when I place the image into the imagemagick directory:

```
magick.exe 169133250-t-4.jpg -gravity center -scale 215x -extent 215x280 -quality 60 RESIZED-169133250-t-4-V.jpg
```

This works fine from the command line when I indicate the actual uploaded image location and resize it etc

```
magick.exe c:\inetpub\vhosts\***.ch\httpdocs\provider_photos\169133250\banner_ads\169133250-t-4.jpg -gravity center -scale 215x -extent 215x280 -quality 60 c:\inetpub\vhosts\***.ch\httpdocs\provider_photos\169133250\photos_thumbnails_states\RESIZED-169133250-t-4-V.jpg
```

I place this into a page using the webdna scripting language

```
[dos]
cd ../
cd ImageMagick
magick.exe c:\inetpub\vhosts\****.ch\httpdocs\provider_photos\[provider_id]\banner_ads\[thumbnail_filename] -gravity center -scale 215x -extent 215x280 -quality 60 c:\inetpub\vhosts\****.ch\httpdocs\provider_photos\[provider_id]\photos_thumbnails_states\[thumbnail_filename]
[/dos]
```

and it fails, well, it does resize etc, but not to what I want to achieve like the 2 command line versions do...
btw, [provider_id] & [[thumbnail_file] have actual info like the command line version, those elements are dynamically added from a db

Why would that happen??

this is all on a Windows server.

Should I add quotes around some parts of the command? Nothing indicates I should..

**Comment by snibgo** [comment]
Your directory name seems to contain an asterisk: `*`. If so, that will create problems.

The forum software interprets an asterisk as markup. Please show your code between two lines that each contain just three backticks: ```

In Windows (and other shells) , if filenames or directory names contain spaces, they need to be quoted.

**Comment by ChiKwasi** [comment, marked as answer]
I didn't notice the docs saying to do this, but I noticed in examples on the site that they were enclosing dimensions in quotes.

I added quotes around 215x280

And it seems to behave as expected.
